contracts/global-extensions/GlobalOptimisticAuctionRebalanceExtension.sol

```diff
@@ -0,0 +1,328 @@
+     * @param _oldComponentsAuctionParams   AuctionExecutionParams for existing components, indexed corresponding to
+     *                                      the current component positions. Set to 0 for components being removed.
+     * @param _shouldLockSetToken           Indicates if the rebalance should lock the SetToken.
+     * @param _rebalanceDuration            Duration of the rebalance in seconds.
+     * @param _positionMultiplier           Position multiplier at the time target units were calculated.
+     */
+    function startRebalance(
+        ISetToken _setToken,
+        IERC20 _quoteAsset,
+        address[] memory _oldComponents,
+        address[] memory _newComponents,
+        AuctionExecutionParams[] memory _newComponentsAuctionParams,
+        AuctionExecutionParams[] memory _oldComponentsAuctionParams,
+        bool _shouldLockSetToken,
+        uint256 _rebalanceDuration,
+        uint256 _positionMultiplier
+    )
+        external
+        override
+    {
+        bytes32 proposalHash = keccak256(abi.encode(
+            _setToken,
+            _quoteAsset,
+            _oldComponents,
+            _newComponents,
+            _newComponentsAuctionParams,
+            _oldComponentsAuctionParams,
+            _shouldLockSetToken,
+            _rebalanceDuration,
+            _positionMultiplier
+        ));
+
+        bytes32 assertionId = assertionIds[proposalHash];
+        require(assertionId != bytes32(0), "Proposal hash does not exist");
+
+        delete assertionIds[proposalHash];
+        delete proposalHashes[assertionId];
+
+        ProductSettings memory settings = productSettings[_setToken];
+        settings.optimisticParams.optimisticOracleV3.settleAndGetAssertionResult(assertionId);
+        
+        address[] memory currentComponents = _setToken.getComponents();
+        
+        require(currentComponents.length == _oldComponents.length, "Mismatch: old and current components length");
+
+        for (uint256 i = 0; i < _oldComponents.length; i++) {
+            require(currentComponents[i] == _oldComponents[i], "Mismatch: old and current components");
+        }
+
+        bytes memory callData = abi.encodeWithSelector(
+            IAuctionRebalanceModuleV1.startRebalance.selector,
+            _setToken,
+            _quoteAsset,
+            _newComponents,
+            _newComponentsAuctionParams,
+            _oldComponentsAuctionParams,
+            _shouldLockSetToken,
+            _rebalanceDuration,
+            _positionMultiplier
+        );
+
+        _invokeManager(_manager((_setToken)), address(auctionModule), callData);
+    }
+
+     // Constructs the claim that will be asserted at the Optimistic Oracle V3.
+    function _constructClaim(bytes32 proposalHash, bytes32 rulesHash) internal pure returns (bytes memory) {
+        return
+            abi.encodePacked(
+                AncillaryData.appendKeyValueBytes32("", PROPOSAL_HASH_KEY, proposalHash),
```

Comment:
Hashing the proposal like this, means that you cannot validate the assertion using this claim alone. (because you cannot deccode the original proposal arguments from the hash).

Do you think this could be a problem for UMAs dispute resolution process ?

Reply:
> Hashing the proposal like this, means that you cannot validate the assertion using this claim alone. (because you cannot deccode the original proposal arguments from the hash).

So long as we emit the settings in an event, this would be easily verifiable with access to the chain.
 
> Do you think this could be a problem for UMAs dispute resolution process ?

This is the route that UMA's team went with for OSnap proposals. A hash of all the transactions while this implementation goes with a hash of the auction settings so I'm thinking it shouldn't be an issue for a disputer to listen for proposals, grab the event, grab the rules and make a decision on disputing.